tiledb/sm/group/group.cc

```diff
@@ -660,21 +660,22 @@ Group::members() const {
   return members_;
 }
 
-Status Group::serialize(Buffer*) {
-  return Status_GroupError("Invalid call to Group::serialize");
+void Group::serialize(Serializer &) {
+  throw StatusException(Status_GroupError("Invalid call to Group::serialize"));
 }
 
-Status Group::apply_and_serialize(Buffer* buff) {
+Status Group::apply_and_serialize(Serializer& serializer) {
   RETURN_NOT_OK(apply_pending_changes());
-  return serialize(buff);
+  serialize(serializer);
+  return Status::Ok();
 }
 
 std::tuple<Status, std::optional<tdb_shared_ptr<Group>>> Group::deserialize(
```

Comment:
We might be able to change every deserialize functions in the review to not return status?

Reply:
Have tried to achieve, removing the tuple<>s containing the Status (until arriving upward in StorageManager::load_group_from_uri() )

Reply:
Great! I think load_group_from_uri is the right stopping point for this PR. Can we also change the functions to return tdb_shared_ptr<Group> instead of std::optional<tdb_shared_ptr<Group>, or is there something I'm not seeing?